### PR TITLE
PLT-311 Add unit test workflow for opt-out-import

### PR DESCRIPTION
**Pull request by gsf**
## 🎫 Ticket

https://jira.cms.gov/browse/PLT-311

## 🛠 Changes

Added workflow for running unit tests on the opt-out-import workflow.

## ℹ️ Context for reviewers

This should run on every PR to check that unit tests pass.

## ✅ Acceptance Validation

See checks for runs. This won't be fully tested until merged and run in another PR.

## 🔒 Security Implications

None.

### CODE REVIEW

**Review by smirnovaae — APPROVED** [review]

File: .github/workflows/opt-out-import-test-unit.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        env:
+          AWS_REGION: ${{ vars.AWS_REGION }}
+        with:
+          params: |
+            ARTIFACTORY_URL=/artifactory/url
+            ARTIFACTORY_USER=/artifactory/user
+            ARTIFACTORY_PASSWORD=/artifactory/password
+      - name: Run unit tests for opt-out-import lambda
+        working-directory: ./optout
+        run: ../gradlew test
```
Comment (smirnovaae) [inline .github/workflows/opt-out-import-test-unit.yml:29]: is it possible to add sonarqube analysis after tests?

Comment (gsf) [reply .github/workflows/opt-out-import-test-unit.yml:29]: Sure I'll do that

Comment (smirnovaae) [reply .github/workflows/opt-out-import-test-unit.yml:29]: One more question, if don't mind
What happens if jenkins build fails and github action pass? Will I able to merge my changes?

Comment (gsf) [reply .github/workflows/opt-out-import-test-unit.yml:29]: For changes to the opt-out-import lambda, there would be no need to run the Jenkins pipeline. The GitHub Actions workflow should run automatically on PR, and the only actual gate to merge is approval from a reviewer.

Comment (gsf) [reply .github/workflows/opt-out-import-test-unit.yml]: Sonar added but see the error in the run linked below. Is there a sonar task in the gradle build file?

https://github.com/CMSgov/ab2d-lambdas/actions/runs/7658938304/job/20872947370?pr=78#step:6:18

Comment (gsf) [reply .github/workflows/opt-out-import-test-unit.yml]: So are you able to run sonar tests specifically for opt-out-import lambda in your dev environment? Which command do you run for that, and in which directory?

Comment (smirnovaae) [reply .github/workflows/opt-out-import-test-unit.yml]: I can't run sonar from `ab2d-lambda/optout`. I assume, working directory should be `ab2d-lambda`.

Locally I've got the same result for `ab2d-lambdas % ./gradlew sonar` and `ab2d-lambdas % ./gradlew sonarrqube`. 
Not sure what's the difference 

I'm fine to hold off sonarqube on the current developing stage

**Comment by gsf** [comment]
Okay I've dropped SonarQube for now and checks are passing. Merging.

**Pull request merged**